src/test/java/software/sigma/bu003/internship/coparts/client/technomir/client/TechnomirClientTest.java

```diff
@@ -0,0 +1,176 @@
+        when(tehnomirClientConfig.getApiToken()).thenReturn(API_TOKEN);
+    }
+
+    @Test
+    void shouldReturnListStockPartsIfSuccessfully() {
+        when(restTemplate.postForObject(
+                URL_TO_API + TechnomirApiUri.PRICE_GET_STOCK_PRICE.getStr(),
+                testTehnomirPayLoad,
+                StockTechnomirPartWrapper.class)).thenReturn(stockTechnomirPartWrapper);
+
+        Optional<List<StockTechnomirPart>> actual = SUT.getListStockParts();
+
+        assertEquals(Optional.of(List.of(stockTechnomirPart)), actual);
+        verify(tehnomirClientConfig, times(1)).getUrlToApi();
+        verify(tehnomirClientConfig, times(1)).getApiToken();
+        verify(restTemplate, times(1)).postForObject(
+                URL_TO_API + TechnomirApiUri.PRICE_GET_STOCK_PRICE.getStr(),
+                testTehnomirPayLoad,
+                StockTechnomirPartWrapper.class);
+    }
+
+    @Test
+    void shouldReturnOptionalEmptyIfAPIReturnEmpty() {
+        when(restTemplate.postForObject(
+                URL_TO_API + TechnomirApiUri.PRICE_GET_STOCK_PRICE.getStr(),
+                testTehnomirPayLoad,
+                StockTechnomirPartWrapper.class)).thenReturn(null);
+
+        Optional<List<StockTechnomirPart>> actual = SUT.getListStockParts();
+
+        assertEquals(Optional.empty(), actual);
+        verify(tehnomirClientConfig, times(1)).getUrlToApi();
+        verify(tehnomirClientConfig, times(1)).getApiToken();
+        verify(restTemplate, times(1)).postForObject(
+                URL_TO_API + TechnomirApiUri.PRICE_GET_STOCK_PRICE.getStr(),
+                testTehnomirPayLoad,
+                StockTechnomirPartWrapper.class);
+    }
+
+    @Test
+    void shouldThrowExceptionIfApiWrong() {
+        when(restTemplate.postForObject(
+                URL_TO_API + TechnomirApiUri.PRICE_GET_STOCK_PRICE.getStr(),
+                testTehnomirPayLoad,
+                StockTechnomirPartWrapper.class)).thenThrow(RestClientException.class);
+
+        assertThrows(TechnomirClientException.class, () -> SUT.getListStockParts());
+        verify(tehnomirClientConfig, times(1)).getUrlToApi();
+        verify(tehnomirClientConfig, times(1)).getApiToken();
+        verify(restTemplate, times(1)).postForObject(
+                URL_TO_API + TechnomirApiUri.PRICE_GET_STOCK_PRICE.getStr(),
+                testTehnomirPayLoad,
+                StockTechnomirPartWrapper.class);
+    }
+
+
+    @Test
+    void shouldReturnPartIfSuccessfully() {
+        testTehnomirPayLoad.setCode(PART_CODE);
+
+        when(restTemplate.postForObject(
+                URL_TO_API + TechnomirApiUri.PRICE_POSITION_SEARCH.getStr(),
+                testTehnomirPayLoad,
+                TechnomirPartWrapper.class)).thenReturn(technomirPartWrapper);
+
+        Optional<List<TechnomirPart>> actual = SUT.getPartsByCode(PART_CODE);
+
+        assertEquals(Optional.of(List.of(technomirPart)), actual);
```

Comment:
this always be true... you don;t modify technomirPart in your client.....

Reply:
corrected